polyfill/lib/ecmascript.mjs

```diff
@@ -981,61 +1001,71 @@ export const ES = ObjectAssign({}, ES2020, {
         value = defaultValue;
       } else {
         any = true;
+        if (BUILTIN_CASTS.has(property)) {
+          value = BUILTIN_CASTS.get(property)(value);
+        }
       }
-      if (BUILTIN_FIELDS.has(property)) {
-        result[property] = ES.ToInteger(value);
-      } else {
-        result[property] = value;
-      }
+      result[property] = value;
     }
     if (!any) {
       throw new TypeError('no supported properties found');
     }
+    // FIXME need a way to indicate that one of {year, eraYear} is required. Should this be delegated to calendars?
```

Comment:
That isn't on the list of things that we decided should be validated outside of the calendar (https://github.com/tc39/proposal-temporal/issues/1229#issuecomment-763056041), so I think it should be delegated to the calendar.